recipe/test_selenium.py

```diff
@@ -0,0 +1,90 @@
+""" test whether firefox is viable with geckodriver and selenium
+
+    Adapted from
+    https://github.com/conda-forge/firefox-feedstock/blob/master/recipe/run_test.py
+"""
+import sys
+import os
+import subprocess
+import traceback
+import json
+
+from pathlib import Path
+
+from selenium import webdriver
+from selenium.webdriver.firefox.options import Options
+
+import pytest
+
+
+@pytest.fixture
+def binary_paths():
+    plat = sys.platform.lower()
+    if "win32" in plat:
+        firefox = Path(os.environ["LIBRARY_BIN"]) / "firefox.exe"
+        geckodriver = Path(os.environ["SCRIPTS"]) / "geckodriver.exe"
+    else:
+        geckodriver = Path(sys.prefix) / "bin" / "geckodriver"
+        app_dir = Path(sys.prefix) / "bin" / "FirefoxApp"
+
+        if "linux" in plat:
+            firefox = app_dir / "firefox"
+        else:
+            firefox = app_dir / "Contents" / "MacOS" / "firefox"
+
+    assert firefox.exists()
+    assert geckodriver.exists()
+
+    return dict(
+        firefox_binary=str(firefox),
+        executable_path=str(geckodriver),
+    )
+
+
+def test_read_license(tmp_path, binary_paths):
+    geckodriver_log = tmp_path / "geckodriver.log"
+    html_log = tmp_path / "license.html"
+    license_png = tmp_path / "license.png"
+
+    print("testing about:license with selenium...")
+    driver = None
+    errors = []
+    try:
+        options = Options()
+        options.headless = True
+        driver = webdriver.Firefox(
+            options=options,
+            service_log_path=str(geckodriver_log),
+            service_args=["--log", "trace"],
+            **binary_paths,
+        )
+        driver.get("about:license")
+
+        if driver.page_source:
+            html_log.write_text(driver.page_source)
+            assert "Mozilla Public License 2.0" in driver.page_source, \
+                "couldn't even load the license page"
+            driver.save_screenshot(str(license_png))
```

Comment:
I'm on the fence about this. Am I right in thinking we wouldn't be able to get the saved png out of the azure box to view it? If yes then we're just testing the save_screenshot capability of geckodriver, which feels out of scope.

But if you feel strongly about it that's fine.

Reply:
screenshots are my bread and butter. also, you _can_ extract them back out from earlier in the log, as they get captured as base64. maybe there's a way to add them as attachments to the xunit... will look into it...

Reply:
Here's how screenshot embedding is done with pytest-html (https://github.com/conda-forge/firefox-feedstock/pull/26#discussion_r463805394), don't know about xunit.

Reply:
interesting, there's some kind of attachments option: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=191234&view=ms.vss-test-web.build-test-results-tab&runId=1146678&resultId=100000&paneView=attachments

Reply:
ah, it's just a very thin wrapper around `pytest-nunit`... will do a staged-recipes for that, but keep the current behavior for now.

Reply:
https://github.com/conda-forge/staged-recipes/pull/12273